src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
+  value < 18 || value > 90
+    ? { validated: false, message: 'Age must be between 18 and 90' }
+    : { validated: true };
+
+const getValidatorFunction = (value, key) => {
+  const validators = {
+    position: positionValidatorFunction,
+    name: nameValidatorFunction,
+    age: ageValidatorFunction,
+  };
+
+  return validators[key]?.(value) || { validated: true };
+};
+
+const validateForm = (data) => {
+  return Array.from(data)
+    .map(([key, value]) => getValidatorFunction(value, key))
+    .filter((result) => !result.validated)
+    .map((result) => result.message);
+};
+
+// NOTIFICATIONS ON SUCESS / ERROR
+const displaySuccessNotice = (message) => {
+  displayNotification('success', 'Success!', [message]);
+};
+
+const displayErrorNotice = (errorArray) => {
+  displayNotification('error', 'Error!', errorArray);
+};
+
+const displayNotification = (type, titleText, errorMessageArr) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.innerHTML = `
+    <h3 class="title">${titleText}</h3>
+    <p>
+      <ul>
+        ${errorMessageArr.map((message) => `<li>${message}</li>`).join('')}
+      </ul>
+    </p>
+  `;
+
+  document.body.append(notification);
+  setTimeout(() => notification.remove(), 2000);
+};
+
+// HANDLE FORM SUBMIT EVENT
+const appendTableWithData = (data) => {
+  const newRow = tBody.insertRow(-1);
+
+  data.forEach((value, key) => {
+    newRow.insertCell(-1).textContent =
+      key === 'salary' ? window.convertNumberToSalary(value) : value;
+  });
+};
+
+const handleSubmitFormEvent = (e) => {
+  e.preventDefault();
+
+  const target = e.target.closest('form');
+  const formData = new FormData(target);
+  const errors = validateForm(formData);
+
+  if (errors.length === 0) {
+    const message = 'New employee has been added to the list!';
+
+    appendTableWithData(formData);
+    displaySuccessNotice(message);
+  } else {
+    displayErrorNotice(errors);
+  }
+};
+
+form.addEventListener('submit', handleSubmitFormEvent);
+// #endregion
+
+// #region CELL_EDITTING
+// EVENT HANDLERS TO SAVE EDITED DATA
+const saveCellInput = (cell, value) => {
+  if (cell instanceof HTMLTableCellElement) {
+    const index = cell.cellIndex;
+    const isSalary = table.rows[0].cells[index].innerHTML === 'Salary';
+
+    cell.innerHTML = isSalary ? window.convertNumberToSalary(value) : value;
+  }
+};
+
+const handleSaveCellEvent = (e) => {
+  const target = e.target;
+  const td = e.target.parentNode;
+
+  if (target && td) {
+    const value = target.value;
+
+    target.hidden = true;
+    saveCellInput(td, value);
+  }
+};
+
+const handleSaveCellOnEnterEvent = (e) => {
+  if (e.key === 'Enter') {
+    e.preventDefault();
+    handleSaveCellEvent(e);
+  }
+};
+
+// CREATE CELL INPUT FIELD
+const createCellInputField = () => {
+  const input = Object.assign(document.createElement('input'), {
```

Comment:
The `createCellInputField` function uses `Object.assign` to set properties on a new input element. This is a bit unusual and may be confusing to other developers. Consider using the regular property assignment syntax instead.